src/controllers/confirm.company.controller.ts

```diff
@@ -45,7 +51,17 @@ export const post = async (req: Request, res: Response, next: NextFunction) => {
     addToObjectionSession(session, SESSION_OBJECTION_ID, objectionId);
     return res.redirect(OBJECTIONS_ENTER_INFORMATION);
   } catch (e) {
-    return next(e);
+    if (e.status === 400 && e.data.status) {
+      const returnTo = getIneligiblePage(e);
```

Comment:
'returnTo' could be better named I think - implies you (the user) has been to this page before or come from it. Maybe just 'redirectTo' or 'ineligiblePage'?